docs/source/reference/benchmark/callback.rst

```diff
@@ -0,0 +1,130 @@
+.. _benchmark-skycallback:
+
+SkyCallback
+===========
+
+SkyCallback is a simple Python library that works in conjunction with SkyPilot Benchmark.
+It enables SkyPilot to provide a more detailed benchmark report without the need to wait until the task finishes.
+
+What SkyCallback is for
+--------------------------------------------
+
+SkyCallback is designed for **machine learning tasks** which have a loop iterating many `steps`.
+SkyCallback measures the average time taken by each step, and extrapolates it to the total execution time of the task.
```

Comment:
How is total execution time extrapolated? Do we know the total number of steps? Good to mention here